client/my-sites/themes/single-site-jetpack.jsx

```diff
@@ -54,6 +69,39 @@ export default connectOptions(
 					<ThanksModal
 						site={ site }
 						source={ 'list' } />
+					{ config.isEnabled( 'manage/themes/upload' ) &&
+						<ThemeUploadCard
+							label={ translate( 'WordPress.com themes' ) }
+						/>
+					}
+					<ThemesSelection
```

Comment:
We'll need to wrap this `ThemesSelection` with a `connectOptions()` call of its own before using it here, I think, and provide a different set of options than the surrounding component gets.